src/common/util.ts

```diff
@@ -87,12 +87,28 @@ export async function findWorkDirectory(
   const dirname = path.dirname(filePath)
   let patterns = [].concat(rootPatterns)
   try {
-    const dir = await findUp(patterns, {
-      cwd: dirname
-    });
+    for(const pattern of patterns) {
+      // attempt to match for directory patterns
+      let dirMatchPath = await findUp(pattern, {
```

Comment:
if patternis `.git`, dirMatchPath is `/path/to/xxx/.git` or `/path/to/xxxx`

Reply:
`dirMatchPath` should be `/path/to/xxx/.git`

Reply:
I cleaned up the matcher so that it should match either a file or a path and then get the enclosing directory starting at `dirname`.